subprojects/configuration-cache/src/test/kotlin/org/gradle/configurationcache/DefaultIgnoredConfigurationInputsTest.kt

```diff
@@ -0,0 +1,131 @@
+    }
+
+    @Test
+    fun `recognizes multiple paths`() {
+        val instance = createFromPaths(listOf("path/one", "path/two"))
+        assertTrue(instance.isFileSystemCheckIgnoredFor(File("path/one")))
+        assertTrue(instance.isFileSystemCheckIgnoredFor(File("path/two")))
+    }
+
+    @Test
+    fun `recognizes path specified with a wildcard but only within one segment`() {
+        val instance = createFromPaths(listOf("path/*"))
+        assertTrue(instance.isFileSystemCheckIgnoredFor(File("path/a")))
+        assertTrue(instance.isFileSystemCheckIgnoredFor(File("path/a.txt")))
+
+        assertFalse(instance.isFileSystemCheckIgnoredFor(File("path/")))
+        assertFalse(instance.isFileSystemCheckIgnoredFor(File("path/two/segments")))
+    }
+
+    @Test
+    fun `recognizes segments that are partially wildcarded`() {
+        val instance = createFromPaths(listOf("foo*/*bar*/*baz/*.xml"))
+        assertTrue(instance.isFileSystemCheckIgnoredFor(File("foo/bar/baz/.xml")))
+        assertTrue(instance.isFileSystemCheckIgnoredFor(File("foo1/2bar3/4baz/abc.xml")))
+    }
+
+    @Test
+    fun `recognizes double-asterisk wildcards across path segments`() {
+        val instance = createFromPaths(listOf("foo/**/bar/**"))
+        assertTrue(instance.isFileSystemCheckIgnoredFor(File("foo/one/two/bar/three")))
+        assertTrue(instance.isFileSystemCheckIgnoredFor(File("foo/2/bar/3/4")))
+    }
+
+    @Test
+    fun `recognizes user-home-based paths against the user home dir`() {
+        val instance = createFromPaths(listOf("~/.gradle/foo.bar"))
+        assertTrue(instance.isFileSystemCheckIgnoredFor(File(System.getProperty("user.home"), ".gradle/foo.bar")))
+    }
+
+    @Test
+    fun `recognizes relative paths pointing outside the root directory`() {
+        val instance = createFromPaths(listOf("../../test1"))
+        assertTrue(instance.isFileSystemCheckIgnoredFor(rootDir.parentFile.parentFile.resolve("test1")))
+    }
+
+    @Test
+    fun `if created as case-sensitive, does not recognize paths in a different case`() {
+        val instance = createFromPaths(listOf("abc", "DEF/*/GHI"), isCaseSensitive = true)
+        assertFalse(instance.isFileSystemCheckIgnoredFor(File("Abc")))
+        assertFalse(instance.isFileSystemCheckIgnoredFor(File("Def/123/ghi")))
+    }
+
+    @Test
+    fun `if created as case-insensitive, recognizes paths in a different case`() {
+        val instance = createFromPaths(listOf("abc", "DEF/*/GHI"), isCaseSensitive = false)
+        assertTrue(instance.isFileSystemCheckIgnoredFor(File("Abc")))
+        assertTrue(instance.isFileSystemCheckIgnoredFor(File("Def/123/ghi")))
+    }
+
+    @Test
+    fun `accepts backslash-separated path patterns on Windows`() {
+        val instance = createFromPaths(listOf("abc\\*\\ghi"))
+        val isRecognized = instance.isFileSystemCheckIgnoredFor(File("abc/test/ghi"))
+        assertEquals(OperatingSystem.current().isWindows, isRecognized)
```

Comment:
❓ Why only on Windows? Would users be losing anything if Gradle allowed backslashes across platforms?

Reply:
This is consistent with the handling of the separators in Java: Windows can use both, while on a non-Windows machine backslashes are regular path characters.

Reply:
Got it.